contracts/token/ERC20/BurnableToken.sol

```diff
@@ -25,21 +25,11 @@ contract BurnableToken is StandardToken {
    * @param _value uint256 The amount of token to be burned
    */
   function burnFrom(address _from, uint256 _value) public {
-    require(_value <= allowed[_from][msg.sender]);
-    // Should https://github.com/OpenZeppelin/zeppelin-solidity/issues/707 be accepted,
-    // this function needs to emit an event with the updated approval.
-    allowed[_from][msg.sender] = allowed[_from][msg.sender].sub(_value);
-    _burn(_from, _value);
+    _burnFrom(_from, _value);
   }
 
   function _burn(address _who, uint256 _value) internal {
-    require(_value <= balances[_who]);
-    // no need to require value <= totalSupply, since that would imply the
-    // sender's balance is greater than the totalSupply, which *should* be an assertion failure
-
-    balances[_who] = balances[_who].sub(_value);
-    totalSupply_ = totalSupply_.sub(_value);
+    super._burn(_who, _value);
```

Comment:
I did this because both `burn` and `burnFrom` need to add the `Burn` event. I felt it was better than adding the event in both. What do you think?

Reply:
That's even worse now that I realize it lol. `burnFrom` calls `_burnFrom`, which calls the overriden `_burn`, which calls the original `burn`: unless you know the details of `StandardToken`, you can't tell that `burnFrom` will emit a `Burn` event. The reason _why_ you did it makes sense though.

I think the problem is the missing comment: if the overridden function said something like 'adds the Burn event when burning', it may read more clearly.